website/docs/r/machine_learning_datastore_fileshare.html.markdown

```diff
@@ -0,0 +1,123 @@
+  name                = "workspace-example-ai"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  application_type    = "web"
+}
+
+resource "azurerm_key_vault" "example" {
+  name                = "workspaceexamplekeyvault"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "premium"
+}
+
+resource "azurerm_storage_account" "example" {
+  name                     = "workspacestorageaccount"
+  location                 = azurerm_resource_group.example.location
+  resource_group_name      = azurerm_resource_group.example.name
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+}
+
+resource "azurerm_machine_learning_workspace" "example" {
+  name                    = "example-workspace"
+  location                = azurerm_resource_group.example.location
+  resource_group_name     = azurerm_resource_group.example.name
+  application_insights_id = azurerm_application_insights.example.id
+  key_vault_id            = azurerm_key_vault.example.id
+  storage_account_id      = azurerm_storage_account.example.id
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_storage_share" "example" {
+  name                 = "example"
+  storage_account_name = azurerm_storage_account.example.name
+  quota                = 1
+}
+
+resource "azurerm_machine_learning_datastore_fileshare" "example" {
+  name                  = "example-datastore"
+  workspace_id          = azurerm_machine_learning_workspace.example.id
+  storage_filesahare_id = azurerm_storage_share.test.resource_manager_id
+  account_key           = azurerm_storage_account.example.primary_access_key
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Machine Learning DataStore. Changing this forces a new Machine Learning DataStore to be created.
+
+* `workspace_id` - (Required) The ID of the Machine Learning Workspace. Changing this forces a new Machine Learning DataStore to be created.
+
+* `storage_fileshare_id` - (Required) The ID of the Storage Account File Share. Changing this forces a new Machine Learning DataStore to be created.
+
+---
+* `account_key` - (Optional) The access key of the Storage Account. Conflicts with `shared_access_signature`.
+
+* `shared_access_signature` - (Optional) The Shared Access Signature of the Storage Account. Conflicts with `account_key`.
+
+* `description` - (Optional) Text used to describe the asset. Changing this forces a new Machine Learning DataStore to be created.
+
+* `is_default` - (Optional) Specifies whether this Machines Learning DataStore is the default for the Workspace. Defaults to `false`.
+
+~> **Note:** `is_default` can only be set to `true` on update.
```

Comment:
As in you cannot set it on creation? could we not just make an update call at the end of the create function to set this to true?

Reply:
This property seems managed by a separate API and not documented in the swagger. For now this property cannot updated. Checking with service team.

Reply:
if its managed by the service then "`is_default` can only be set to `true` on update." is not valid?

and then maybe we should just make it computed?

Reply:
Have changed `is_default`  to computed